src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyContentsAdminListDisplayDriver.cs

```diff
@@ -46,8 +51,18 @@ public override async Task<IDisplayResult> EditAsync(ContentOptionsViewModel mod
                 return null;
             }
 
+            var taxonomyContentItemIds = settings.TaxonomyContentItemIds;
+            if (!string.IsNullOrWhiteSpace(model.SelectedContentType))
+            {
+                var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(model.SelectedContentType);
+                var fieldDefinitions = contentTypeDefinition
+                    .Parts.SelectMany(x => x.PartDefinition.Fields.Where(f => f.FieldDefinition.Name == nameof(TaxonomyField)));
+                var fieldTaxonomyContentItemIds = fieldDefinitions.Select(x => x.GetSettings<TaxonomyFieldSettings>().TaxonomyContentItemId);
+                taxonomyContentItemIds = taxonomyContentItemIds.Intersect(fieldTaxonomyContentItemIds).ToArray();
+            }
+
             var results = new List<IDisplayResult>();
-            var taxonomies = await _contentManager.GetAsync(settings.TaxonomyContentItemIds);
+            var taxonomies = await _contentManager.GetAsync(taxonomyContentItemIds);
```

Comment:
These changes seem unrelated to the issue. The problem is that you're merging from your fork's dev branch; it's better to create a branch for every patch you're working on. As a stopgap solution, you can just revert the commit where you've made this change.

Reply:
Restored